MobiusCore/Source/Mobius.swift

```diff
@@ -45,36 +39,36 @@ public extension Mobius {
     ///   - update: the `Update` function of the loop
     ///   - effectHandler: an instance conforming to the `ConnectableProtocol`. Will be used to handle effects by the loop
     /// - Returns: a `Builder` instance that you can further configure before starting the loop
-    static func loop<T: LoopTypes, C: Connectable>(update: @escaping Update<T>, effectHandler: C) -> Builder<T> where C.InputType == T.Effect, C.OutputType == T.Event {
-        return Builder<T>(
+    static func loop<Model, Event, Effect, C: Connectable>(update: @escaping Update<Model, Event, Effect>, effectHandler: C) -> Builder<Model, Event, Effect> where C.InputType == Effect, C.OutputType == Event {
+        return Builder(
             update: update,
             effectHandler: effectHandler,
             initiator: { First(model: $0) },
-            eventSource: AnyEventSource<T.Event>({ _ in AnonymousDisposable(disposer: {}) }),
+            eventSource: AnyEventSource({ _ in AnonymousDisposable(disposer: {}) }),
             eventQueue: DispatchQueue(label: "event processor"),
             effectQueue: DispatchQueue(label: "effect processor", attributes: .concurrent),
-            logger: AnyMobiusLogger(NoopLogger<T>())
+            logger: AnyMobiusLogger(NoopLogger())
```

Comment:
Here is an example of type inference that wasn't possible before, as well as the `eventSource` line above.